src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -0,0 +1,180 @@
+            }
+            // pass the token into the AuthenticationManager system
+            // this indirectly calls GuardAuthenticationProvider::authenticate()
+            $token = $this->authenticationManager->authenticate($token);
+
+            if (null !== $this->logger) {
+                $this->logger->info('Guard authentication successful!', array('token' => $token, 'authenticator' => get_class($guardAuthenticator)));
+            }
+
+            // sets the token on the token storage, etc
+            $this->guardHandler->authenticateWithToken($token, $request);
+        } catch (AuthenticationException $e) {
+            // oh no! Authentication failed!
+
+            if (null !== $this->logger) {
+                $this->logger->info('Guard authentication failed.', array('exception' => $e, 'authenticator' => get_class($guardAuthenticator)));
+            }
+
+            $response = $this->guardHandler->handleAuthenticationFailure($e, $request, $guardAuthenticator);
+
+            if ($response instanceof Response) {
+                $event->setResponse($response);
+            }
+
+            return;
+        }
+
+        // success!
+        $response = $this->guardHandler->handleAuthenticationSuccess($token, $request, $guardAuthenticator, $this->providerKey);
+        if ($response instanceof Response) {
+            if (null !== $this->logger) {
+                $this->logger->info('Guard authenticator set success response.', array('response' => $response, 'authenticator' => get_class($guardAuthenticator)));
+            }
+
+            $event->setResponse($response);
+        } else {
+            if (null !== $this->logger) {
+                $this->logger->info('Guard authenticator set no success response: request continues.', array('authenticator' => get_class($guardAuthenticator)));
+            }
+        }
+
+        // attempt to trigger the remember me functionality
+        $this->triggerRememberMe($guardAuthenticator, $request, $token, $response);
+    }
+
+    /**
+     * Should be called if this listener will support remember me.
+     *
+     * @param RememberMeServicesInterface $rememberMeServices
+     */
+    public function setRememberMeServices(RememberMeServicesInterface $rememberMeServices)
+    {
+        $this->rememberMeServices = $rememberMeServices;
+    }
+
+    /**
+     * Checks to see if remember me is supported in the authenticator and
+     * on the firewall. If it is, the RememberMeServicesInterface is notified.
+     *
+     * @param GuardAuthenticatorInterface $guardAuthenticator
+     * @param Request                     $request
+     * @param TokenInterface              $token
+     * @param Response                    $response
+     */
+    private function triggerRememberMe(GuardAuthenticatorInterface $guardAuthenticator, Request $request, TokenInterface $token, Response $response = null)
+    {
+        if (!$guardAuthenticator->supportsRememberMe()) {
+            return;
```

Comment:
Should you not inform that the guard doesn't support rembember-me? And do this check *after* checking if remember-me was actually enabled for the firewall.